torch/fx/graph.py

```diff
@@ -440,39 +426,40 @@ def output(self, result: Argument, type_expr: Optional[Any] = None):
         """
         return self.create_node(op='output', target='output', args=(result,), type_expr=type_expr)
 
-    def _name(self, target: Target) -> str:
+    def _target_to_str(self, target : Target) -> str:
         if callable(target):
             op = target.__name__
         else:
             assert isinstance(target, str)
             op = target
             if _is_magic(op):
                 op = op[2:-2]
-        op = op.replace('.', '_')
+        return op
+
+    def _create_unique_name(self, candidate : str) -> str:
+        candidate = candidate.replace('.', '_')
         # delete all characters that are illegal in a Python identifier
-        op = re.sub('[^0-9a-zA-Z_]+', '_', op)
-        op = _snake_case(op)
-        if op[0].isdigit():
-            op = f'_{op}'
-
-        return self._register_name_used(op)
-
-    def _register_name_used(self, op : str) -> str:
-        """
-        Even if a user provides us with a name, we must register that that
-        name is used to prevent duplication of names from further nodes as
-        well as ensure that the name provided does not shadow a builtin.
-        """
-        if op not in self._used_names:
-            self._used_names[op] = 0
-            # Avoid shadowing PyTorch and Python builtins.
-            if not hasattr(torch, op) and \
-               not hasattr(torch.nn.functional, op) and \
-               not hasattr(torch.nn, op) and \
-               not _shadows_builtin_name(op):
-                return op
-        i = self._used_names[op] = self._used_names[op] + 1
-        return f'{op}_{i}'
+        candidate = re.sub('[^0-9a-zA-Z_]+', '_', candidate)
+        candidate = _snake_case(candidate)
```

Comment:
snake casing should probably go in _target_to_string. We want to take names as is when they are valid identifiers if they were provided as part of the `create_node` API.